### PR TITLE
Provide provider_job_id to qu adapter.

### PR DESCRIPTION
**Pull request by kddnewton**
Further work to provide provider_job_id for queue adapters.

### CODE REVIEW

**Comment by rafaelfranca** [comment]
Don't forget the CHANGELOG

**Comment by kddeisz** [comment]
@rafaelfranca - thanks! Just updated it.

**Comment by rafaelfranca** [comment]
Tests seems to be broken.

**Comment by kddeisz** [comment]
It's not displaying the travis errors for me, do you have a link? It was broken until about 20 minutes ago - is it still now? I had to add the "unless qu_job.nil?" because of the different backends.

**Comment by kddeisz** [comment]
@rafaelfranca any idea how to proceed? Both the regular and integration tests are passing for me, so I don't exactly know what to do. Thanks.

**Comment by kddeisz** [comment]
Thanks @matthewd - looks like it's good now.

File: activejob/test/integration/queuing_test.rb [hunk 1]
```diff
@@ -56,13 +56,16 @@ class QueuingTest < ActiveSupport::TestCase
...
-    skip unless adapter_is?(:sidekiq) || adapter_is?(:que) || adapter_is?(:delayed_job)
+  test 'should supply a provider_job_id when available for immediate jobs' do
+    skip unless adapter_is?(:delayed_job, :sidekiq, :qu, :que)
     test_job = TestJob.perform_later @id
-    refute test_job.provider_job_id.nil?, "Provider job id should be set by provider"
+    refute test_job.provider_job_id.nil?, 'Provider job id should be set by provider'
+  end
 
+  test 'should supply a provider_job_id when available for delayed jobs' do
+    skip unless adapter_is?(:delayed_job, :sidekiq, :que)
```
Comment (jvanbaarsen) [inline activejob/test/integration/queuing_test.rb]: I think you forgot to add the provider_job_id for `qu` in the enqueed_at method.

Comment (kddeisz) [inline activejob/test/integration/queuing_test.rb]: Yeah qu doesn't support scheduled jobs.

Comment (jvanbaarsen) [inline activejob/test/integration/queuing_test.rb]: Ah sorry, I only saw the first line of ` def enqueue_at(job, timestamp, *args) #:nodoc:` did not read the full method body.

**Pull request merged**